### DISCUSSION TITLE
Your thoughts: SMA Emulation Tool

### DISCUSSION THREAD

**Discussion by nurF** [SBFspot/SBFspot#571] [Ideas]
Hi all,

let me shortly explain my idea here.
The SMA eco system lacks a way to integrate third party devices.
In my specific case I would like to make a Victron Quattro visible within my SMA based home world.
You can find some rationale [here](https://community.victronenergy.com/questions/156133/victron-integration-into-the-sma-world.html) and [here](https://www.photovoltaikforum.com/thread/182468-victron-integration-in-die-sma-welt/).

I have checked the source code of this project and I find this very usable for my purpose,
as I have a very strong C / C++ background under Windows and Linux.
Obviously the SBFspot tool is intended to be a reader/monitor tool and no writer/emulator.
But I see a lot of reusable/shareable code between SBFspot and the tool I am planning to create.

**Feedback**
I would like to get some feedback from the core developers of this project:
What do you think about my intentions?
I could add another executable that is created as a second artifact from all build steps.
I am still at the very beginning, so I am very open and flexible right now.

Best,
nurF

**Comment by theoprivat** [comment]
Hello, that's exactly what I want to do. Does anyone already have something going on here?

Best,
Theo
